File: Iterator/Javascript/UserIterator.js
import IIterator from './IIterator.js';

class UserIterator extends IIterator {
  constructor(users) {
    super();
    this.users = users;
    this.index = 0;
  }

  hasNext() {
    return this.index < this.users.length;
  }

  next() {
    if (this.hasNext()) {
      return this.users[this.index++];
    } else if (this.users.length > 0) {
      return this.users[this.users.length - 1];
    }
  }
}

export default UserIterator;

File: Iterator/Python 3/UserIterator.py
from IIterator import IIterator

class UserIterator(IIterator):
  def __init__(self, users):
    self.users = users
    self.index = 0
  
  def hasNext(self):
    return self.index < len(self.users)
  
  def next(self):
    if self.hasNext():
      user = self.users[self.index]
      self.index += 1
      return user
    elif len(self.users) > 0:
      return self.users[len(self.users) - 1]